app/models/info_request.rb

```diff
@@ -165,7 +165,7 @@ class << self
   validate :must_be_valid_state
   validates_inclusion_of :prominence, :in => Prominence::VALUES
 
-  validates_inclusion_of :law_used, in: Legislation.keys
+  validates_inclusion_of :law_used, in: Legislation.keys, allow_nil: true
```

Comment:
I don't think we want to allow a `nil` or `blank` value to be saved.

We have the database constraint of course, which the model validations should mirror. If we don't do this, there's a possibility of the app raising a `PG::NotNullViolation` instead of gracefully handling the issue.

Reply:
Once #6339 this can be reverted as this is to allow specs to pass

Reply:
The specs this fixes are:

```
Failures:

  1) RequestController#new_batch when batch requests is enabled when the current user can make batch requests should render 'preview' when given a good title and body
     Failure/Error: expect(response).to render_template('preview')
       expecting <"preview"> but rendering with <["request/new", "layouts/default"]>
     # ./spec/controllers/request_controller_spec.rb:2008:in `block (4 levels) in <top (required)>'
     # ./.direnv/gems/2.6.7p197/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  2) RequestController#new_batch when batch requests is enabled when the current user can make batch requests on success sets the batch_sent flash to true
     Failure/Error: expect(flash[:batch_sent]).to be true

       expected true
            got nil
     # ./spec/controllers/request_controller_spec.rb:2059:in `block (5 levels) in <top (required)>'
     # ./.direnv/gems/2.6.7p197/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  3) RequestController#new_batch when batch requests is enabled when the current user can make batch requests on success should create an info request batch and redirect to the new batch on success
     Failure/Error: expect(new_info_request_batch).not_to be_nil

       expected: not nil
            got: nil
     # ./spec/controllers/request_controller_spec.rb:2045:in `block (5 levels) in <top (required)>'
     # ./.direnv/gems/2.6.7p197/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  4) RequestController#new_batch when batch requests is enabled when the current user can make batch requests on success should prevent double submission of a batch request
     Failure/Error: expect(assigns[:existing_batch]).not_to be_nil

       expected: not nil
            got: nil
     # ./spec/controllers/request_controller_spec.rb:2054:in `block (5 levels) in <top (required)>'
     # ./.direnv/gems/2.6.7p197/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

What is happening is this conditional is true since teh `@info_request` has no law used since the default value was removed in the previous commit in this PR
https://github.com/mysociety/alaveteli/blob/b1d999103889f6754d78d49e92cfad41e3302017/app/controllers/request_controller.rb#L245-L252